ext/opentelemetry-exporter-cloud-trace/src/opentelemetry/tools/resource_detector.py

```diff
@@ -0,0 +1,46 @@
+import requests
+
+from opentelemetry.context import attach, set_value
+from opentelemetry.sdk.resources import Resource, ResourceDetector
+
+_GCP_METADATA_URL = (
+    "http://metadata.google.internal/computeMetadata/v1/?recursive=true"
+)
+_GCP_METADATA_URL_HEADER = {"Metadata-Flavor": "Google"}
+
+
+def get_gce_resources():
+    """ Resource finder for common GCE attributes
+
+        See: https://cloud.google.com/compute/docs/storing-retrieving-metadata
+    """
+    attach(set_value("suppress_instrumentation", True))
+    all_metadata = requests.get(
+        _GCP_METADATA_URL, headers=_GCP_METADATA_URL_HEADER
+    ).json()
+    gce_resources = {
+        "host.id": all_metadata["instance"]["id"],
+        "cloud.account.id": all_metadata["project"]["projectId"],
+        "cloud.zone": all_metadata["instance"]["zone"].split("/")[-1],
+        "cloud.provider": "gcp",
+        "gcp.resource_type": "gce_instance",
+    }
+    return gce_resources
+
+
+_RESOURCE_FINDERS = [get_gce_resources]
+
+
+class GoogleCloudResourceDetector(ResourceDetector):
```

Comment:
I'm assuming this will exist in a not trace specific package if monitoring and trace exporters are merged? Right now, do users have to import this package even if they just want resources for monitoring?

Reply:
Ah yeah someone else flagged this too, tracked here https://github.com/GoogleCloudPlatform/opentelemetry-operations-python/issues/12